docs/reference/api/platform/ATCmdParser.md

```diff
@@ -1,11 +1,37 @@
 ## ATCmdParser
 
-[Add description here]
+ATCmdParser is an Mbed OS compatible AT command parser. 
+
+It sends and receives AT commands to and from a communication device implementing the [FileHandle](./FileHandle.md) interface. It also implements AT command parsing, which validates the data format and separate command and data portion of AT transactions. For example, the UARTSerial communication driver implements the FileHandle interface, and you can use it with ATCmdParser to send and receive AT commands. The actual format of AT commands used depends on the communication device used.
+
+To use ATCmdParser, a reference of object implementing [FileHandle](./FileHandle.md) interface is passed ATCmdParser constructor during ATCmdParser object creation. ATCmdParser also supports configuring specific output delimiter character sequence depending on the interface or device connected to the communication interface.
```

Comment:
Query: Who or what passes ATCmdParser constructor during ATCmdParser object creation?

Reply:
Good catch, let me fix the comments and make a push.